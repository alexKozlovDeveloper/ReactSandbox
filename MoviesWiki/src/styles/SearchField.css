.searchfield {
    font-family: cursive;
    box-shadow: inset 0 1px 3px #ddd;
    border: 1px solid #ccc;
    padding-left: 6px;
    padding-right: 6px;
    padding-top: 6px;
    padding-bottom: 6px;
    width: 50%;
}