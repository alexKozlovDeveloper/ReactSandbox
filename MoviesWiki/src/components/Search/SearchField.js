import React, { Component } from "react";

import styles from "../../styles/SearchField.css";

const SearchField = ({ placeHolder, updateFunc, submitFunc }) => {
    return <input className={styles.searchfield} type="text" placeholder={placeHolder} onInput={(e) => updateFunc(e.currentTarget.value)} onKeyDown={(e) => submitFunc(e)} />;
}

export default SearchField;