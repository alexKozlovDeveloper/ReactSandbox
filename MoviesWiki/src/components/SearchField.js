import React, { Component } from "react";

import styles from "../styles/SearchField.css";

function SearchField(props) {
    return  <input className={styles.searchfield} type="text" placeholder={props.placeHolder} />;
}

export default SearchField;