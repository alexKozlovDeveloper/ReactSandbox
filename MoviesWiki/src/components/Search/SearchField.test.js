import React from 'react';
import { shallow, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import SearchField from "./SearchField";

configure({ adapter: new Adapter() });

test('SearchField rendering test', () => {

    var placeholder = 'some placeholder';

    const component = shallow(<SearchField placeholder={placeholder} />);
    expect(shallowToJson(component)).toMatchSnapshot();
});

