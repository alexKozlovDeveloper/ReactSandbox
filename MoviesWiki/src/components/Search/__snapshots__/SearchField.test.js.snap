// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchField rendering test 1`] = `
<input
  className="searchfield"
  type="text"
/>
`;
